package/MDAnalysis/analysis/base.py

```diff
@@ -163,3 +167,151 @@ def run(self):
         logger.info("Finishing up")
         self._conclude()
         return self
+
+
+class AnalysisFromFunction(AnalysisBase):
+    """
+    Create an analysis from a function working on AtomGroups
+
+    Attributes
+    ----------
+    results : ndarray
+        results of calculation are stored after call to ``run``
+
+    Example
+    -------
+    >>> def rotation_matrix(mobile, ref):
+    >>>     return mda.analysis.align.rotation_matrix(mobile, ref)[0]
+
+    >>> rot = AnalysisFromFunction(rotation_matrix, trajectory, mobile, ref).run()
+    >>> print(rot.results)
+
+    Raises
+    ------
+    ValueError : if ``function`` has the same kwargs as ``BaseAnalysis``
+    """
+
+    def __init__(self, function, trajectory=None, *args, **kwargs):
+        """Parameters
+        ----------
+        function : callable
+            function to evaluate at each frame
+        trajectory : mda.coordinates.Reader (optional)
+            trajectory to iterate over. If ``None`` the first AtomGroup found in
+            args and kwargs is used as a source for the trajectory.
+        *args : list
+           arguments for ``function``
+        **kwargs : dict
+           arugments for ``function`` and ``AnalysisBase``
+
+        """
+        if (trajectory is not None) and (not isinstance(
+                trajectory, coordinates.base.Reader)):
+            args = args + (trajectory, )
+            trajectory = None
+
+        if trajectory is None:
+            for arg in args:
+                if isinstance(arg, AtomGroup):
+                    trajectory = arg.universe.trajectory
+            # when we still didn't find anythgin
+            if trajectory is None:
+                for arg in six.itervalues(kwargs):
+                    if isinstance(arg, AtomGroup):
+                        trajectory = arg.universe.trajectory
+
```

Comment:
I made it explicit now. Should be better then some deep stack that has to be inspected.